File: app/src/main/java/com/instar/frontend_android/ui/activities/LoginEmailActivity.java
package com.instar.frontend_android.ui.activities;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.text.InputType;
import android.view.View;
import android.view.inputmethod.InputMethodManager;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageButton;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import com.instar.frontend_android.R;
import com.instar.frontend_android.ui.customviews.ViewEditText;
import com.instar.frontend_android.ui.customviews.ViewEffect;

public class LoginEmailActivity extends AppCompatActivity {
    private ImageButton imageBack;
    private TextView btnText;

    private TextView title;
    private TextView textNote;
    private View layout;
    private View layoutEmail;
    private EditText emailText;
    private TextView labelEmail;
    private ImageButton btnRemove;
    private Button btnFindAccount;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login_email);
        imageBack = findViewById(R.id.imageBack);
        layout = findViewById(R.id.layout);
        layoutEmail = findViewById(R.id.email);
        emailText = layoutEmail.findViewById(R.id.editText);
        labelEmail = layoutEmail.findViewById(R.id.textView);
        btnRemove = layoutEmail.findViewById(R.id.btnImage);
        title =  findViewById(R.id.emailTitle).findViewById(R.id.Title);
        textNote =  findViewById(R.id.emailTitle).findViewById(R.id.textNote);
        btnText = findViewById(R.id.emailTitle).findViewById(R.id.btnText);
        btnFindAccount = findViewById(R.id.btnFindAccount);
        loadActivity();
        initView();
    }
    public void loadActivity() {
        title.setText("Tìm tài khoản");
        textNote.setText("Nhập tên người dùng, email hoặc số di động của bạn");
        btnText.setText("Bạn không thể đặt lại mật khẩu?");
        labelEmail.setText("Tên người dùng, email/số di động");
        emailText.setHint("Tên người dùng, email/số di động");
        btnFindAccount.setText("Tìm tài khoản");
    }
    public void initView() {
        ViewEditText viewEditText = new ViewEditText();
        viewEditText.EditTextRemove(layoutEmail,emailText,labelEmail,btnRemove);
        ViewEffect.ViewText(btnText);
        btnText.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

            }
        });
        btnFindAccount.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(LoginEmailActivity.this, LoginSmsActivity.class);
                startActivity(intent);
            }
        });
        layout.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                InputMethodManager imm = (InputMethodManager) getSystemService(Context.INPUT_METHOD_SERVICE);
                imm.hideSoftInputFromWindow(v.getWindowToken(), 0);
                emailText.clearFocus();
                if (emailText.getText().toString().isEmpty()) {
                    setEmail();
                }
            }
        });
        imageBack.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                finish();
            }
        });
        effectClick();
    }

    public void setEmail() {
        labelEmail.setVisibility(View.GONE);
        layoutEmail.setBackground(getDrawable(R.drawable.border_component_login_dow));
        emailText.setHint("Tên người dùng, email/số di động");
    }

    public void effectClick() {
        ViewEffect.ImageBack(imageBack);
    }
}

File: app/src/main/java/com/instar/frontend_android/ui/activities/LoginEmailActivity.kt
package com.instar.frontend_android.ui.activities

import android.annotation.SuppressLint
import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.View
import android.view.inputmethod.InputMethodManager
import android.widget.Button
import android.widget.EditText
import android.widget.ImageButton
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.instar.frontend_android.R
import com.instar.frontend_android.databinding.ActivityLoginEmailBinding
import com.instar.frontend_android.databinding.EdittextLoginBinding
import com.instar.frontend_android.types.requests.LoginRequest
import com.instar.frontend_android.types.requests.VerifyCodeRequest
import com.instar.frontend_android.types.responses.ApiResponse
import com.instar.frontend_android.ui.customviews.ViewEditText
import com.instar.frontend_android.ui.customviews.ViewEffect
import com.instar.frontend_android.ui.services.AuthService
import com.instar.frontend_android.ui.services.ServiceBuilder
import com.instar.frontend_android.ui.services.ServiceBuilder.handleResponse
import com.instar.frontend_android.ui.utils.Helpers
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class LoginEmailActivity : AppCompatActivity() {
    private lateinit var binding: ActivityLoginEmailBinding

    private lateinit var imageBack: ImageButton
    private lateinit var btnText: TextView
    private lateinit var title: TextView
    private lateinit var textNote: TextView
    private lateinit var layout: View
    private lateinit var layoutEmail: EdittextLoginBinding
    private lateinit var emailText: EditText
    private lateinit var labelEmail: TextView
    private lateinit var btnRemove: ImageButton
    private lateinit var btnFindAccount: Button
    private lateinit var authService: AuthService


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        authService = ServiceBuilder.buildService(AuthService::class.java, this)

        setContentView(binding.root)

        imageBack = binding.imageBack
        layout = binding.layout
        layoutEmail = binding.email
        emailText = layoutEmail.editText
        labelEmail = layoutEmail.textView
        btnRemove = layoutEmail.btnImage
        title = binding.emailTitle.Title
        textNote = binding.emailTitle.textNote
        btnText = binding.emailTitle.btnText
        btnFindAccount = binding.btnFindAccount

        loadActivity()
        initView()
    }

    private fun loadActivity() {
        title.text = "Tìm tài khoản"
        textNote.text = "Nhập tên người dùng, email hoặc số di động của bạn"
        btnText.text = "Bạn không thể đặt lại mật khẩu?"
        labelEmail.text = "Tên người dùng, email/số di động"
        emailText.hint = "Tên người dùng, email/số di động"
        btnFindAccount.text = "Tìm tài khoản"
    }

    private fun initView() {
        val viewEditText = ViewEditText()
        viewEditText.EditTextRemove(layoutEmail.Layout, emailText, labelEmail, btnRemove)
        ViewEffect.ViewText(btnText)
        btnText.setOnClickListener {
        }

        btnFindAccount.setOnClickListener {
            if (!Helpers.isValidEmail(emailText.text.toString()) || emailText.text.toString().isEmpty()) {
                return@setOnClickListener;
            }

            val verifyCodeRequest = VerifyCodeRequest().apply {
                email = emailText.text.toString()
            }

            authService.verifyCode(verifyCodeRequest).handleResponse(
                onSuccess = { response ->
                    // Handle successful response
                    val intent = Intent(this@LoginEmailActivity, ResetPasswordActivity::class.java)
                    intent.putExtra("email", verifyCodeRequest.email)
                    startActivity(intent)
                },
                onError = { error ->
                    // Handle error
                    Log.e("ServiceBuilder", "Error: $error")
                }
            )

        }

        layout.setOnClickListener {
            val imm = getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
            imm.hideSoftInputFromWindow(it.windowToken, 0)
            emailText.clearFocus()
            if (emailText.text.toString().isEmpty()) {
                setEmail()
            }
        }

        imageBack.setOnClickListener { finish() }

        effectClick()
    }

    @SuppressLint("UseCompatLoadingForDrawables")
    private fun setEmail() {
        labelEmail.visibility = View.GONE
        layoutEmail.Layout.background = getDrawable(R.drawable.border_component_login_dow)
        emailText.hint = "Tên người dùng, email/số di động"
    }

    private fun effectClick() {
        ViewEffect.ImageBack(imageBack)
    }
}